lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -323,48 +419,111 @@ export default function getRecipeInterface(
             return SessionFunctions.getSessionInformation(helpers, sessionHandle);
         },
 
+        /*
+            In all cases: if sIdRefreshToken token exists (so it's a legacy session) we clear it.
+            Check http://localhost:3002/docs/contribute/decisions/session/0008 for further details and a table of expected behaviours
+         */
         refreshSession: async function (
             this: RecipeInterface,
-            { req, res }: { req: BaseRequest; res: BaseResponse }
+            { req, res, userContext }: { req: BaseRequest; res: BaseResponse; userContext: any }
         ): Promise<Session> {
             logDebugMessage("refreshSession: Started");
-            let inputIdRefreshToken = getIdRefreshTokenFromCookie(req);
-            if (inputIdRefreshToken === undefined) {
-                logDebugMessage("refreshSession: UNAUTHORISED because idRefreshToken from cookies is undefined");
-                // we do not clear cookies here because of a
-                // race condition mentioned here: https://github.com/supertokens/supertokens-node/issues/17
 
-                throw new STError({
-                    message: "Session does not exist. Are you sending the session tokens in the request as cookies?",
-                    type: STError.UNAUTHORISED,
-                });
+            // This token isn't handled by getToken/setToken to limit the scope of this legacy/migration code
+            if (req.getCookieValue(LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME) !== undefined) {
+                setCookie(config, res, LEGACY_ID_REFRESH_TOKEN_COOKIE_NAME, "", 0, "accessTokenPath");
+            }
```

Comment:
should this only happen if we were successfully able to refresh the incoming session?

Reply:
There are 3 cases (involving the id refresh token):
- We get the id refresh token but no refresh token for some reason: we can remove it because future refreshes will fail  anyway
- We get the id refresh token and a refresh token, but the refresh fails for some reason:
  - anti-csrf check fails: we can keep it
  - core refresh call fails: we clear all tokens, so we should remove this as well
- We get the id refresh token, and the refresh token and the refresh succeeds: we should remove it because it was upgraded

I suppose I can move this around to avoid clearing if only the anti-csrf check fails.